### DISCUSSION TITLE
What is the algorithm used by `reconstruct_surface`?

### DISCUSSION THREAD

**Discussion by stla** [pyvista/pyvista#2149] [Q&A] [answered]
Hello,

  The C++ library [CGAL](https://github.com/CGAL/cgal/issues/6224) has some difficulties to reconstruct the surface of the Hopf torus from a points cloud.  This works fine with PyVista's `reconstruct surface` (see link above). Do you know which algorithm it uses?

**Comment by banesullivan** [comment, marked as answer]
The `reconstruct_surface` filter utilizes [`vtkSurfaceReconstructionFilter`](https://vtk.org/doc/nightly/html/classvtkSurfaceReconstructionFilter.html#details):

> [vtkSurfaceReconstructionFilter](https://vtk.org/doc/nightly/html/classvtkSurfaceReconstructionFilter.html) takes a list of points assumed to lie on the surface of a solid 3D object. A signed measure of the distance to the surface is computed and sampled on a regular grid. The grid can then be contoured at zero to extract the surface. The default values for neighborhood size and sample spacing should give reasonable results for most uses but can be set if desired. This procedure is based on the PhD work of Hugues Hoppe: http://www.research.microsoft.com/~hoppe

I'm not sure if the algorithm has a name, but I imagine you may have some luck understanding this looking into Hugues Hoppe's research

**Comment by stla** [comment]
Did you change something to `reconstruct_surface`? Now the reconstruction of the Hopf torus is ugly:

![hopfscrewed](https://user-images.githubusercontent.com/4466543/165123194-b94cd2e0-636e-4ef1-93f7-fb7eba6f4fe3.png)

I'm pretty sure I used the same data.
